test/register-bson.js

```diff
@@ -1,7 +1,65 @@
 'use strict';
 
-// TODO(NODE-3555): When we remove karma in favor of vm we can use ts-node to test src instead of lib here.
-const BSON = require('../lib/bson');
-const { ensureBuffer } = require('../lib/ensure_buffer');
-BSON.ensureBuffer = ensureBuffer;
+require('source-map-support').install({
+  hookRequire: true
+});
+
+const { inspect } = require('node:util');
+const vm = require('node:vm');
+const fs = require('node:fs');
+const path = require('node:path');
+const chai = require('chai');
+
+/**
+ * In the runInContext "web" testing instanceof checks fail
+ * since the error classes are declared in a different realm (?)
+ * So here we augment the chai assertion to fallback on checking the name property of the error instance
+ */
+chai.use(function (chai) {
+  const throwsAssertion = chai.Assertion.prototype.throw;
+  chai.Assertion.addMethod('throw', function (...args) {
+    try {
+      throwsAssertion.call(this, ...args);
+    } catch (assertionError) {
+      if (assertionError.actual?.name === assertionError.expected) {
+        return;
+      }
+      throw assertionError;
+    }
+  });
+});
+
+// Register expect globally, this is no longer necessary since we do not use karma / bundle everything together
+globalThis.expect = chai.expect;
+
+// Controls whether to run BSON library declaration in an node or "web" environment
+const web = process.env.WEB === 'true';
+console.error(inspect({ web }, { colors: true }));
+
+// Node.js 16 was when the atob and btoa globals were introduced, so we need replacements for testing on 14
+const shim_btoa = input => Buffer.prototype.toString.call(Buffer.from(input), 'base64');
+const shim_atob = input => Buffer.from(input, 'base64').toString('binary');
```

Comment:
Instead of figuring out skip logic on node 14 it's easier to throw these shims on the context object

Reply:
can you add a TODO to remove when we drop node 14 in V6? actually, can you file that drop 14 ticket in the BSON V6 epic - you can name it generically to "drop unsupported node versions" but then link it here; it'll make clean up either if we identify version-specific logic ahead of time

Reply:
done!